### PR TITLE
Ignore multi docs

### PR DESCRIPTION
**Pull request by qiq**

### CODE REVIEW

File: chart_updater/updater.py [hunk 1]
```diff
@@ -40,20 +40,34 @@ def _build_commit_message(manifest: Manifest) -> str:
...
-            manifest.load(path)
-            if manifest.update_with_latest_chart(self.helm_repo):
-                manifest.save(path)
-                commit_message = self._build_commit_message(manifest)
-                self.git.update_file(path, commit_message)
-                updated = True
+        updated = any(
+            self._update_manifest(path)
+                for path in self._manifests_to_check()
+        )
```
Comment (KPilnacek) [inline chart_updater/updater.py]: `any` is about to end the iteration when it encounters the first `True` from the iterator. Either put a `list` around `self._update_manifest(path) for path in self._manifests_to_check()` or iterate over it the standard way.

File: chart_updater/updater.py [hunk 2]
```diff
@@ -40,20 +40,34 @@ def _build_commit_message(manifest: Manifest) -> str:
...
+        try:
+            manifest = Manifest(self.annotation_prefix)
+            manifest.load(path)
+            if not manifest.update_with_latest_chart(self.helm_repo):
+                return False
+            manifest.save(path)
+            commit_message = self._build_commit_message(manifest)
+            self.git.update_file(path, commit_message)
+            return True
+        except UpdateException as e:
```
Comment (KPilnacek) [inline chart_updater/updater.py:58]: consider having the try except only around the part that can actually raise the `UpdateException`

File: tests/test_chart_sync.py [hunk 3]
```diff
@@ -381,6 +405,20 @@ def test_chart_updated_manifest_inside_path(empty_git_repo, requests_mock):
...
+    _add_manifest(MANIFEST_WITH_MULTIPLE_DOCUMENTS, path="deploy/1-multi.yaml")
+    _add_manifest(MANIFEST_WITH_SINGLE_IMAGE, path="deploy/2-helmrelease.yaml")
+    _init_commit()
+    requests_mock.get(HELM_REPO_INDEX, text=CHART_REPO_INDEX_WITH_NEW_CHARTS)
+
+    updater = Updater(Git(empty_git_repo), HelmRepo(HELM_REPO_URL))
+    updater.update_loop(one_shot=True)
+
+    assert _get_manifest("deploy/2-helmrelease.yaml") == UPDATED_MANIFEST_WITH_SINGLE_IMAGE
+    assert re.search(SINGLE_IMAGE_RELEASE_COMMIT_RE, _last_commit())
```
Comment (KPilnacek) [inline tests/test_chart_sync.py:416]: consider also checking that the `MANIFEST_WITH_MULTIPLE_DOCUMENTS` was not changed

File: chart_updater/updater.py [hunk 4]
```diff
@@ -42,7 +42,8 @@ def _build_commit_message(manifest: Manifest) -> str:
 
     def _manifests_to_check(self) -> Iterator[str]:
         return (
-            path for path in self.git.grep(self.annotation_prefix + "/")
+            path for path in self.git.grep("HelmRelease")
+                if self.git.grep(self.annotation_prefix + "/", path)
```
Comment (KPilnacek) [inline chart_updater/updater.py]: consider doing sets of "HelmRelease" and "self.annotation_prefix" greps and iterating only over their intersection

**Review by KPilnacek — CHANGES_REQUESTED** [review]
except that `any` it LGTM

File: chart_updater/updater.py (hunk 1 above)
Comment (qiq) [reply chart_updater/updater.py]: Added list to prevent short-circuit of manifest evaluation.

File: chart_updater/updater.py (hunk 2 above)
Comment (qiq) [reply chart_updater/updater.py:58]: I am not sure it helps the code to be better readable:
```
            manifest = Manifest(self.annotation_prefix)
            try:
                manifest.load(path)
                if not manifest.update_with_latest_chart(self.helm_repo):
                    return False
                manifest.save(path)
            except UpdateException as e:
                log.info(str(e))
                return False
            commit_message = self._build_commit_message(manifest)
            self.git.update_file(path, commit_message)
            return True
```

File: tests/test_chart_sync.py (hunk 3 above)
Comment (qiq) [reply tests/test_chart_sync.py:416]: OK, added, see fixup.

File: chart_updater/updater.py (hunk 4 above)
Comment (qiq) [reply chart_updater/updater.py]: OK, added, see fixup.

File: chart_updater/updater.py (hunk 1 above)
Comment (KPilnacek) [reply chart_updater/updater.py]: I don't see the change

File: chart_updater/updater.py (hunk 2 above)
Comment (KPilnacek) [reply chart_updater/updater.py:58]: Maybe if the 
```
                manifest.load(path)
                if not manifest.update_with_latest_chart(self.helm_repo):
                    return False
                manifest.save(path)
```
is extracted

Comment (qiq) [reply chart_updater/updater.py:58]: Still, not sure it helps:
```
    def _update_with_latest_chart(self, manifest: Manifest, path: str) -> bool:
        try:
            manifest.load(path)
            if not manifest.update_with_latest_chart(self.helm_repo):
                return False
            manifest.save(path)
            return True
        except UpdateException as e:
            log.info(str(e))
            return False

    def _update_manifest(self, path: str) -> bool:
            manifest = Manifest(self.annotation_prefix)
            updated = self._update_with_latest_chart(manifest, path)
            if not updated:
                return False
            commit_message = self._build_commit_message(manifest)
            self.git.update_file(path, commit_message)
            return True
```

Comment (qiq) [reply chart_updater/updater.py:58]: I can imagine UpdateException is triggered in _build_commit_message or git.update_file as well.

Comment (KPilnacek) [reply chart_updater/updater.py:58]: I had in mind sth like:
```python
    def _update_manifest_file(self, manifest):
        manifest.load(path)
        if not manifest.update_with_latest_chart(self.helm_repo):
            raise NoUpdateException
        manifest.save(path)

    def _update_manifest(self, path: str) -> bool:
        manifest = Manifest(self.annotation_prefix)
        try:
            self._update_manifest_file(manifest)
        except NoUpdateException:
            return False
        except UpdateException as e:
            log.info(str(e))
            return False
        commit_message = self._build_commit_message(manifest)
        self.git.update_file(path, commit_message)
        return True
```

**Review by KPilnacek — APPROVED** [review]

**Pull request merged**